lib/streamlit/runtime/pages_manager.py

```diff
@@ -0,0 +1,147 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import threading
+from pathlib import Path
+from typing import Callable, Final
+
+from blinker import Signal
+
+import streamlit.source_util as source_util
+from streamlit.logger import get_logger
+from streamlit.util import calc_md5
+from streamlit.watcher import watch_dir
+
+_LOGGER: Final = get_logger(__name__)
+
+
+class V1PagesManager:
```

Comment:
nit: this name should probably be something more like `V1PagesWatcher` since the class just installs watchers/doesn't do anything else related to page management

Reply:
nit: I would also suggest to flip the word order and call it `PagesManagerV1`. This way, file sorting etc. shows the different versions next to each other